lib/rust/ensogl/core/src/application/shortcut.rs

```diff
@@ -235,83 +235,122 @@ pub struct Registry {
         let mouse = mouse.clone_ref();
         let command_registry = command_registry.clone_ref();
         let shortcuts_registry = default();
-        Self { keyboard, mouse, command_registry, shortcuts_registry, currently_handled }
+        Self { mouse, command_registry, shortcuts_registry, currently_handled, target }
     }
 
-    fn process_rules(&self, rules: &[Shortcut]) {
+    fn process_rules(&self, stop_propagation: impl FnOnce<()>, rules: &[Shortcut]) {
         let mut targets = Vec::new();
         {
             let borrowed_command_map = self.command_registry.name_map.borrow();
+            let bound_target =
+                self.target.and_then(|id| self.command_registry.id_map.borrow().get(&id).cloned());
             for rule in rules {
-                let target = &rule.action.target;
-                borrowed_command_map.get(target).for_each(|instances| {
-                    for instance in instances {
-                        if Self::condition_checker(&rule.condition, &instance.status_map) {
-                            let command_name = &rule.command.name;
-                            match instance.command_map.borrow().get(command_name) {
-                                Some(cmd) =>
-                                    if cmd.enabled {
-                                        targets.push((cmd.frp.clone_ref(), command_name))
-                                    },
-                                None => warn!("Command {command_name} was not found on {target}."),
-                            }
+                let instances = match bound_target.as_ref() {
+                    Some(target) => slice::from_ref(target),
+                    None => borrowed_command_map
+                        .get(&rule.action.target)
+                        .map(Vec::as_slice)
+                        .unwrap_or_default(),
+                };
+                for instance in instances {
+                    if Self::condition_checker(&rule.condition, &instance.status_map) {
+                        let command_name = &rule.command.name;
+                        match instance.command_map.borrow().get(command_name) {
+                            Some(cmd) =>
+                                if cmd.enabled {
+                                    targets.push((cmd.frp.clone_ref(), command_name))
+                                },
+                            None => error!(
+                                "Command {command_name} was not found on {}.",
+                                match bound_target.as_ref() {
+                                    Some(_) => "bound instance",
+                                    None => &rule.action.target,
+                                }
+                            ),
                         }
                     }
-                })
+                }
             }
         }
-
+        if !targets.is_empty() {
+            warn!("- stop_propagation");
```

Comment:
This warning is a bit obscure.

Reply:
Oops. I had removed that in a small commit that I forgot to push.